addons/sale/models/sale_order_line.py

```diff
@@ -420,16 +420,20 @@ def _get_custom_compute_tax_cache_key(self):
 
     @api.depends('product_id', 'product_uom', 'product_uom_qty')
     def _compute_pricelist_item_id(self):
-        for line in self:
-            if not line.product_id or line.display_type or not line.order_id.pricelist_id:
-                line.pricelist_item_id = False
-            else:
-                line.pricelist_item_id = line.order_id.pricelist_id._get_product_rule(
-                    line.product_id,
-                    line.product_uom_qty or 1.0,
-                    uom=line.product_uom,
-                    date=line._get_order_date(),
-                )
+        no_item_lines = self.filtered(
+            lambda rec: not rec.product_id or rec.display_type or not rec.order_id.pricelist_id
+        )
+        other_lines = self - no_item_lines
+        no_item_lines.update({"pricelist_item_id": False})
```

Comment:
```suggestion
        no_item_lines.pricelist_item_id = False
```

Field assignment works on multi-records sets and is cleaner IMHO.

Reply:
fixed